iml-agent/src/action_plugins/kernel_module.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, cmd::cmd_output};
+
+pub async fn loaded(module: String) -> Result<bool, ImlAgentError> {
+    let output = cmd_output("lsmod", vec![]).await?;
+
+    if output.status.success() {
+        let stdout = String::from_utf8_lossy(&output.stdout);
+        let mut modules = stdout.lines().into_iter().filter_map(|m| {
+            let mut fields = m.split(' ').filter(|s| *s != "");
+            let name = fields.next();
+            if let Some(name) = name {
+                Some(name)
+            } else {
+                None
+            }
+        });
+
+        let module = modules.find(|m| *m == module);
```

Comment:
I think here you don't need to walk over the list/iterator again. ~~Checking for non-empty is enough, because above you have already filtered out non-matching lines.~~ You can compare with `Some(module)` in `filter_map` in one pass, or even use `find` to make it less than one pass :)

Reply:
Yep, done